executors/cmdexec/volume.go

```diff
@@ -285,6 +285,11 @@ func (s *CmdExecutor) VolumeInfo(host string, volume string) (*executors.Volume,
 		return nil, fmt.Errorf("Unable to determine volume info of volume name: %v", volume)
 	}
 	logger.Debug("%+v\n", volumeInfo)
+	if volumeInfo.OpErrStr != "" {
```

Comment:
Have to ensure that gluster volume info output always have empty string for success cases. Otherwise, there is no way to determine whether the op failed or succeeded. 

I will do a quick check and update here.

Reply:
I have tried the general scenario. It looks fine to me. I am mainly interested in faulty one how that can be reproduced? So we can check the exact behavior there.

[root@dhcp47-119 ~]# gluster v info test1 >/dev/null
Volume test1 does not exist

[root@dhcp47-119 ~]# gluster v info test1 2>/dev/null

[root@dhcp47-119 ~]# gluster v info test >/dev/null

[root@dhcp47-119 ~]# gluster v info test 2>/dev/null 
Volume Name: test
Type: Replicate
Volume ID: 8e7f0e6d-7a6b-4aa1-b60f-d74efa54d93a
Status: Started
Snapshot Count: 0
Number of Bricks: 1 x 4 = 4
Transport-type: tcp
Bricks:
Brick1: dhcp47-139.lab.eng.blr.redhat.com:/bricks/brick1/b1
Brick2: dhcp47-69.lab.eng.blr.redhat.com:/bricks/brick1/b1
Brick3: dhcp46-149.lab.eng.blr.redhat.com:/bricks/brick1/b1
Brick4: dhcp47-119.lab.eng.blr.redhat.com:/bricks/brick1/b1
Options Reconfigured:
ganesha.enable: on
features.cache-invalidation: on
storage.fips-mode-rchecksum: on
transport.address-family: inet
nfs.disable: on
performance.client-io-threads: off
nfs-ganesha: enable
cluster.enable-shared-storage: enable

Reply:
@iamniting Thanks for the output for all the cases.